### DISCUSSION TITLE
how to send css while sending component as json ?

### DISCUSSION THREAD

**Discussion by geminigeek** [tailwindlabs/tailwindcss#8965] [Help] [answered]
hi,

i am serving dynamic/async components in vue 3, sending components  via json like below in runtime , i already have a tailwind/postcss generated css file for application, but for my dynamic components as i send below, i want to send css/stylesheet along the json !
can anyone help me how to build and send only css required for class key in json below.
  
```
{
    "component":"MyButton",
    "class":"text-2xl font-semibold text-red-400",
    "props": {"label":"Click me"},
    "styles":"here i want to send css styles generated by class key above",
    "stylesheet":"or here i want to send css stylesheet file url to embed which was generated from above class key "
}
```

thanks
Dee

**Comment by wongjn** [comment, marked as answer]
Tailwind isn't really equipped for incremental builds like this, but what I think is that you would have to run Tailwind every time you send through one of these JSON packets. Instead of using `tailwind.config.js`, you could run `postcss` with the Tailwind plugin and pass the Tailwind config inline. This gives you the chance to send the classes directly:

```js
const postcss = require('postcss');
const tailwind = require('tailwindcss');

/**
 * Generates Tailwind CSS from given class names.
 *
 * @param {string} classNames Space-separated list of strings.
 * @return {LazyResult} The process result.
 */
module.exports = classNames => postcss([tailwind({ safelist: classNames.split(' ') })])
  .process('@tailwind base; @tailwind components; @tailwind utilities');
```

(Not sure if this code would actually run, but something like this! Taken from [PostCSS JS API documentation](https://postcss.org/api/))

You could also see if Tailwind has a JavaScript API and do something similar without the need for PostCSS as an intermediary.

**Reply by geminigeek** [reply]
Hi, 
thanks for your reply, i was able to build classes with some thing like you suggested, i am thinking about running a diff for incremental styles after generating styles like you suggested , i think it should work.

Dee
